File: 14. Average Filtering/Manual.m
img = imread('a.tif');
%Manual Average Filter with low values
h = ones(5, 5)/25;
img_manual = imfilter(img, h);
%Manual Average Filter with high values
h = ones(15, 15)/225;
img_manual2 = imfilter(img, h);
figure('Name', 'Original');
imshow(img);
figure('Name', 'Manual Average Blur1');
imshow(img_manual);
figure('Name', 'Manual Average Blur2');
imshow(img_manual2);

File: 14. Average Filtering/Manual.py
import cv2
import numpy as np

img = cv2.imread('cameraman.tif', 0)
h = np.ones((5, 5), np.float64)/25
k = cv2.filter2D(img, -1, h)
cv2.imshow("Original", img)
cv2.imshow("Manual Average", k)
